src/transformers/generation_utils.py

```diff
@@ -457,7 +457,7 @@ def _prepare_attention_mask_for_generation(
         pad_token_id: int,
         eos_token_id: int,
     ) -> torch.LongTensor:
-        is_input_ids = isinstance(inputs, torch.LongTensor) and len(inputs.shape) == 2
```

Comment:
this test was bad and doesn't work. It's better to directly check `inputs.dtype`